dissect/target/plugins/apps/texteditor/windowsnotepad.py

```diff
@@ -0,0 +1,377 @@
+                options = c_windowstab.options_v1(fh).dumps()
+            elif tab_header.optionsVersion == 2:
+                options = c_windowstab.options_v2(fh).dumps()
+            else:
+                # Raise an error, since we don't know how many bytes future optionVersions will occupy.
+                # Now knowing how many bytes to parse can mess up the alignment and structs.
+                raise Exception("Unknown option version")
+
+            # If the file is not saved to disk and no fixedSizeBlockLength is present, an extra checksum stub
+            # is present. So parse that first
+            if file_header.fileType == 0 and tab_header.fixedSizeBlockLength == 0:
+                # Two unknown bytes before the CRC32
+                tab_header_crc32_stub = c_windowstab.tab_header_crc32_stub(fh)
+
+                # Calculate CRC32 of the header and check if it matches
+                actual_header_crc32 = _calc_crc32(
+                    file_header.dumps()[3:] + tab_header.dumps() + options + tab_header_crc32_stub.dumps()[:-4]
+                )
+                if tab_header_crc32_stub.crc32 != actual_header_crc32:
+                    logging.warning(
+                        "CRC32 mismatch in header of file: %s (expected=%s, actual=%s)",
+                        file.name,
+                        tab_header_crc32_stub.crc32.hex(),
+                        actual_header_crc32.hex(),
+                    )
+
+            # Used to store the final content
+            content = ""
+
+            # After a fixed_size_data_block, some more variable_size_data_blocks can be present. This boolean
+            # keeps track of whether more data is still present.
+            has_remaining_data = False
+
+            # In the case that a fixedSizeDataBlock is present, this value is set to a nonzero value
+            if tab_header.fixedSizeBlockLength > 0:
+                # So we parse the fixed size data block
+                data_entry = c_windowstab.fixed_size_data_block(fh)
+
+                # The header (minus the magic) plus all data is included in the checksum
+                actual_crc32 = _calc_crc32(
+                    file_header.dumps()[3:] + tab_header.dumps() + options + data_entry.dumps()[:-4]
+                )
+
+                if data_entry.crc32 != actual_crc32:
+                    logging.warning(
+                        "CRC32 mismatch in single-block file: %s (expected=%s, actual=%s)",
+                        file.name,
+                        data_entry.crc32.hex(),
+                        actual_crc32.hex(),
+                    )
+
+                # Add the content of the fixed size data block to the tab content
+                content += data_entry.data
+
+                # The hasRemainingVariableDataBlocks indicates whether more data will follow after this single block
+                if data_entry.hasRemainingVariableDataBlocks == 1:
```

Comment:
Cant you just use this value to test for remaining data? Instead of using the `has_remaining_data` variable.